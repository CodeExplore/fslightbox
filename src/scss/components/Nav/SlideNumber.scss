.fslightbox-slide-number-container {
  height: 100%;
  font-size: 14px;
  color: #d7d7d7;
  z-index: 0;
  max-width: 46px;

  div {
    padding: 0 2px;
  }
}

.fslightbox-slash {
  padding-top: .15rem;
}