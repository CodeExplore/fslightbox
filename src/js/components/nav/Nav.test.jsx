import React from 'react';
import { shallow } from 'enzyme';
import Nav from "./Nav";

const fsLightbox = { data: { sourcesCount: 2 } };
let nav;

test('data.sourcesCount > 2 - rendering SlideNumber and testing DOM', () => {
    fsLightbox.data.sourcesCount = 2;
    nav = shallow(<Nav fsLightbox={ fsLightbox }/>);

    expect(nav.find('SlideNumber')).toHaveLength(1);
});

test('data.sourcesCount === 1 - not rendering SlideNumber', () => {
    fsLightbox.data.sourcesCount = 1;
    nav = shallow(<Nav fsLightbox={ fsLightbox }/>);

    expect(nav.find('SlideNumber')).toHaveLength(0);
});
