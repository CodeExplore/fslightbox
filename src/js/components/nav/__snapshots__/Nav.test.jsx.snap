// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav DOM should match snapshot 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={Object {}}
  />
  <SlideNumber
    fsLightbox={Object {}}
  />
</div>
`;

exports[`data.sourcesCount > 2 - rendering SlideNumber and testing DOM 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={
      Object {
        "data": Object {
          "sourcesCount": 2,
        },
      }
    }
  />
  <SlideNumber
    fsLightbox={
      Object {
        "data": Object {
          "sourcesCount": 2,
        },
      }
    }
  />
</div>
`;

exports[`data.sourcesCount > 2 - rendering SlideNumber and testing DOM should match snapshot 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={
      Object {
        "data": Object {
          "sourcesCount": 2,
        },
      }
    }
  />
  <SlideNumber
    fsLightbox={
      Object {
        "data": Object {
          "sourcesCount": 2,
        },
      }
    }
  />
</div>
`;

exports[`data.sourcesCount > 2 - rendering SlideNumber and testing DOM should match snapshow 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={
      Object {
        "data": Object {
          "sourcesCount": 2,
        },
      }
    }
  />
  <SlideNumber
    fsLightbox={
      Object {
        "data": Object {
          "sourcesCount": 2,
        },
      }
    }
  />
</div>
`;

exports[`nav DOM should match snapshot 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={Object {}}
  />
  <SlideNumber
    fsLightbox={Object {}}
  />
</div>
`;

exports[`testing DOM should match snapshot 1`] = `
<div
  className="fslightbox-nav"
>
  <Toolbar
    fsLightbox={Object {}}
  />
  <SlideNumber
    fsLightbox={Object {}}
  />
</div>
`;
